src/components/TimeCardPage/CellTypes/CommentCell.tsx

```diff
@@ -1,43 +1,223 @@
+    return isEditing ? (
+      <ButtonGroup justifyContent="center" size="sm">
+        <Button leftIcon={<CheckIcon />} {...getSubmitButtonProps()} />
+        <Button leftIcon={<CloseIcon />} {...getCancelButtonProps()} />
+      </ButtonGroup>
+    ) : (
+      <Flex justifyContent="right">
+        <Button size="sm" leftIcon={<EditIcon />} {...getEditButtonProps()} />
+      </Flex>
+    );
+  };
+
+  const elevatedUserPrivileges =
+    user.Type === "Supervisor" || user.Type === "Admin";
+
+  // probably need to add some more validation
+  const addNewComment = (value: string) => {
+    if (value !== "") {
+      setDisplayedComment(createNewComment(user, displayedComment.Type, value));
+    }
+  };
+
+  const saveComment = () => {
+    setComments([...comments, displayedComment]);
+    // TODO: save to DB
+  };
+
+  // make the editable work as intended later, without the odd preview box and whatever
+  return (
+    <>
+      <IconButton
+        colorScheme={color}
+        aria-label="Report"
+        icon={icon}
+        onClick={onOpen}
+      />
+      <Modal isOpen={isOpen} onClose={onClose}>
+        <ModalOverlay />
+        <ModalContent>
+          <ModalHeader>{displayedComment.Type}</ModalHeader>
+          <ModalCloseButton />
+          <ModalBody>
+            <Editable
+              isDisabled={!isEditable}
+              defaultValue={displayedComment.Content}
+              onSubmit={(value) => addNewComment(value)}
+            >
+              <Input as={EditableInput} />
+              <EditablePreview />
+
+              {elevatedUserPrivileges && (
+                <>
+                  <EditableTextarea />
+                  <EditableControls />
+                </>
+              )}
+            </Editable>
+          </ModalBody>
+
+          <ModalFooter>
+            <Button colorScheme={Color.Blue} mr={3} onClick={onClose}>
+              Close
+            </Button>
+            {isEditable && elevatedUserPrivileges && (
+              <Button colorScheme={Color.Green} onClick={saveComment}>
+                Save
+              </Button>
+            )}
+          </ModalFooter>
+        </ModalContent>
+      </Modal>
+    </>
+  );
+}
+
+const getAllCommentsOfType = (type, commentArray) => {
+  if (commentArray === undefined) {
+    return [];
+  } else {
+    return commentArray.filter((comment) => comment.Type === type);
+  }
+};
+
+export function CommentCell(props: CommentProps) {
+  const [comments, setComments] = useState(
+    getAllCommentsOfType(CommentType.Comment, props.comments)
+  );
+  const [reports, setReports] = useState(
+    getAllCommentsOfType(CommentType.Report, props.comments)
+  );
+  const [isEditable, setisEditable] = useState(false);
+  const user = useContext(UserContext);
+
+  useEffect(() => {
+    //Supervisor/Admins have the right to edit comments/reports
+    if (user.Type === "Supervisor" || user.Type === "Admin") {
+      setisEditable(true);
```

Comment:
Everything here looks good - same feedback from the other ticket regarding this.